### PR TITLE
Resolve staticcheck linter reported issues

### PR DESCRIPTION
**Pull request by chaunceyjiang**
Signed-off-by: chaunceyjiang <chaunceyjiang@gmail.com>

Thank you for contributing to Velero!

# Please add a summary of your change

Please see https://golang.org/doc/go1.16#ioutil

Resolve staticcheck linter reported issues.

# Does your change fix a particular issue?

Part of #5734
# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by blackpiglet** [comment]
@chaunceyjiang 
Thanks for your contribution, but there are still some issues found in the GitHub action check. Please help to fix them.

And there are some issues with the PR. Are those issues ignored intentionally?

```
{"code":"S1000","severity":"error","location":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/cmd/velero-restore-helper/velero-restore-helper.go","line":35,"column":2},"end":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/cmd/velero-restore-helper/velero-restore-helper.go","line":49,"column":3},"message":"should use for range instead of for { select {} }"}
{"code":"S1007","severity":"error","location":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/hack/release-tools/chk_version.go","line":31,"column":36},"end":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/hack/release-tools/chk_version.go","line":31,"column":188},"message":"should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice"}
{"code":"U1000","severity":"error","location":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/hook/item_hook_handler_test.go","line":43,"column":6},"end":{"file":"","line":0,"column":0},"message":"type mockItemHookHandler is unused"}
{"code":"U1000","severity":"error","location":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/hook/item_hook_handler_test.go","line":47,"column":31},"end":{"file":"","line":0,"column":0},"message":"func (*mockItemHookHandler).HandleHooks is unused"}
{"code":"ST1005","severity":"error","location":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/hook/wait_exec_hook_handler.go","line":122,"column":11},"end":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/hook/wait_exec_hook_handler.go","line":122,"column":106},"message":"error strings should not be capitalized"}
{"code":"ST1005","severity":"error","location":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/hook/wait_exec_hook_handler.go","line":158,"column":13},"end":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/hook/wait_exec_hook_handler.go","line":158,"column":111},"message":"error strings should not be capitalized"}
{"code":"ST1005","severity":"error","location":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/hook/wait_exec_hook_handler.go","line":197,"column":11},"end":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/hook/wait_exec_hook_handler.go","line":197,"column":98},"message":"error strings should not be capitalized"}
{"code":"ST1005","severity":"error","location":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/hook/wait_exec_hook_handler.go","line":215,"column":11},"end":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/hook/wait_exec_hook_handler.go","line":215,"column":150},"message":"error strings should not be capitalized"}
{"code":"S1008","severity":"error","location":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/storage/storagelocation.go","line":74,"column":2},"end":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/storage/storagelocation.go","line":76,"column":3},"message":"should use 'return !time.Now().UTC().Before(nextValidation)' instead of 'if time.Now().UTC().Before(nextValidation) { return false }; return true'"}
{"code":"SA1019","severity":"error","location":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/storage/storagelocation_test.go","line":166,"column":14},"end":{"file":"/Users/jxun/go/src/github.com/vmware-tanzu/velero/internal/storage/storagelocation_test.go","line":166,"column":42},"message":"fake.NewFakeClientWithScheme is deprecated: Please use NewClientBuilder instead. "}
{"code":"compile","severity":"error","location":{"file":"pkg/restore/restore.go","line":24,"column":2},"end":{"file":"","line":0,"column":0},"message":"package io/os is not in GOROOT (/usr/local/go/src/io/os)"}
```

**Comment by chaunceyjiang** [comment]
> but there are still some issues found in the GitHub action check. Please help to fix them.

Thanks for reminding.

> And there are some issues with the PR. Are those issues ignored intentionally?

 I will continue to complete the work.

File: pkg/restore/restore.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
-	"io/ioutil"
+	"io/os"
```
Comment (blackpiglet) [inline pkg/restore/restore.go:24]: This line should change to `"os"`

**Comment by blackpiglet** [comment]
The detailed content of the Golang release note deprecated the `io/ioutil` package.

```
Deprecation of io/ioutil
The [io/ioutil](https://go.dev/pkg/io/ioutil/) package has turned out to be a poorly defined and hard to understand collection of things. All functionality provided by the package has been moved to other packages. The io/ioutil package remains and will continue to work as before, but we encourage new code to use the new definit[io](https://go.dev/pkg/io/)ns in the io and [os](https://go.dev/pkg/os/) packages. Here is a list of the new locations of the names exported by io/ioutil:

[Discard](https://go.dev/pkg/io/ioutil/#Discard) => [io.Discard](https://go.dev/pkg/io/#Discard)
[NopCloser](https://go.dev/pkg/io/ioutil/#NopCloser) => [io.NopCloser](https://go.dev/pkg/io/#NopCloser)
[ReadAll](https://go.dev/pkg/io/ioutil/#ReadAll) => [io.ReadAll](https://go.dev/pkg/io/#ReadAll)
[ReadDir](https://go.dev/pkg/io/ioutil/#ReadDir) => [os.ReadDir](https://go.dev/pkg/os/#ReadDir) (note: returns a slice of [os.DirEntry](https://go.dev/pkg/os/#DirEntry) rather than a slice of [fs.FileInfo](https://go.dev/pkg/io/fs/#FileInfo))
[ReadFile](https://go.dev/pkg/io/ioutil/#ReadFile) => [os.ReadFile](https://go.dev/pkg/os/#ReadFile)
[TempDir](https://go.dev/pkg/io/ioutil/#TempDir) => [os.MkdirTemp](https://go.dev/pkg/os/#MkdirTemp)
[TempFile](https://go.dev/pkg/io/ioutil/#TempFile) => [os.CreateTemp](https://go.dev/pkg/os/#CreateTemp)
[WriteFile](https://go.dev/pkg/io/ioutil/#WriteFile) => [os.WriteFile](https://go.dev/pkg/os/#WriteFile)
```

**Comment by chaunceyjiang** [comment]
Thanks for clarifying.

> https://github.com/vmware-tanzu/velero/actions/runs/3870289880/jobs/6597112903
> PR 5744 is missing a changelog.

Does this PR need to write a changelog?

@blackpiglet

**Comment by blackpiglet** [comment]
@chaunceyjiang 
Better to have a changelog file. After all, a lot of files are modified in this PR.

**Comment by chaunceyjiang** [comment]
Hi @blackpiglet 
This pr only solves some of the issues in https://github.com/vmware-tanzu/velero/issues/5734, 
I suggest that we can merge this pr first, and gradually solve https://github.com/vmware-tanzu/velero/issues/5734

what do u think?

**Comment by blackpiglet** [comment]
Sure. I agree with that.

**Review by blackpiglet — DISMISSED** [review]

**Comment by blackpiglet** [comment]
Sorry about creating two commits by accident in this PR.
I already backed out of change, but the merge main commit looks harmless, so I keep it by now. If that is not expected, please change as your wish.

**Pull request closed without merging**